lib/did_you_mean/spell_checkers/name_error_checkers/class_names.rb

```diff
@@ -0,0 +1,37 @@
+# frozen-string-literal: true
+require 'delegate'
+
+module DidYouMean
+  module ClassNames
```

Comment:
I'm not sure if this is a good name - I would probably just change the `DidYouMean::VariableNameChecker` to depend on the `DidYouMean::ClassNameChecker` for class name suggestions. Something along the lines of:

```diff
--- a/lib/did_you_mean/spell_checkers/name_error_checkers/variable_name_checker.rb
+++ b/lib/did_you_mean/spell_checkers/name_error_checkers/variable_name_checker.rb
@@ -2,9 +2,11 @@

 require "did_you_mean/spell_checker"

+require_relative './class_name_checker'
+
 module DidYouMean
   class VariableNameChecker
-    attr_reader :name, :method_names, :lvar_names, :ivar_names, :cvar_names
+    attr_reader :name, :method_names, :lvar_names, :ivar_names, :cvar_names, :class_name_checker

     NAMES_TO_EXCLUDE = { 'foo' => [:fork, :for] }
     NAMES_TO_EXCLUDE.default = []
@@ -71,12 +73,15 @@ module DidYouMean
       @ivar_names   = receiver.instance_variables
       @cvar_names   = receiver.class.class_variables
       @cvar_names  += receiver.class_variables if receiver.kind_of?(Module)
+
+      @class_name_checker = ClasssNameChecker.new(exception)
     end

     def corrections
       @corrections ||= SpellChecker
                      .new(dictionary: (RB_RESERVED_WORDS + lvar_names + method_names + ivar_names + cvar_names))
                      .correct(name) - NAMES_TO_EXCLUDE[@name]
+                     + class_name_checker.corrections
     end
   end
 end
```

(of course this is untested)